nixos/modules/services/mail/maddy.nix

```diff
@@ -215,6 +214,39 @@ in {
         '';
       };
 
+      tls = {
+        loader = mkOption {
+          type = types.enum [ "file" "off" ];
+          default = "off";
+          description = lib.mdDoc ''
+            TLS certificates are obtained by modules called "certificate
+            loaders". Currently only the file loader is supported which reads
+            certificates from files specifying the options `keyPaths` and
+            `certPaths`.
+          '';
+        };
+
+        keyPaths = mkOption {
```

Comment:
Why don't we go with a sub module in the style of `services.madd.tls.certificates = [{ key = "/var/lib/ssl/mydomain.key"; cert = "/var/lib/ssl/mydomain.cert"; }]`here?

I don't exactly know how / if list ordering is handles / preserved when merging multiple definitions, but I think having two lists whose values are only used linked with each other, is more fragile and less intuitive compared to `listOf submodule`.

Reply:
I don't know of any explicitly using this style, but I also don't know any module doing it the proposed way I think.
All modules I think of have a 1:1 mapping of Domain / instance / … to Certificates. But since maddy doesn't do that, we sadly can't do the same.